server/api/user/update/service.js

```diff
@@ -0,0 +1,146 @@
+import has from 'lodash/has';
+import isEmpty from 'lodash/isEmpty';
+import {
+  InvalidPrimaryEmailError,
+  DuplicateEmailAddressError,
+  DuplicateUsernameError,
+  InvalidUsernameError,
+} from '../../../constants/errors';
+import deleteUserPicture from '../deletePicture/service';
+import commonNames from '../../../utils/commonNames';
+
+async function updateUser({ db, storage }, user, nextProps) {
+  const UserModel = db.model('User');
+  const CredentialsModel = db.model('Credentials');
+
+  const nextUser = {};
+  const nextCredentials = {};
+
+  // decorate nextUser obj with emails
+  if (nextProps.emails) {
+    // ensure there is exactly 1 primary email
+    const primaryEmails = nextProps.emails.filter((email) => email.isPrimary);
+    if (primaryEmails.length < 1) {
+      throw new InvalidPrimaryEmailError('You must specify at least 1 primary email');
+    }
+    if (primaryEmails.length > 1) {
+      throw new InvalidPrimaryEmailError('User cannot have more than 1 primary emails');
+    }
+
+    // if requestor changes the previous primary email it also needs to be verified
+    const primaryEmail = primaryEmails[0];
+    const emailExisted = user.emails.find((email) => primaryEmail.address === email.address);
+    if (!emailExisted || !emailExisted.isVerified) {
```

Comment:
I made the decision to not allow a superuser to set a non-verified email as primary also (also the same case for the user) unless they also provide the `isVerified: true` in the same request. This was to try and defend the code a little. If i move the entire check to the `isUserAllowed` function inside the handler then we can probably enable this as well. Which would you prefer?

Reply:
Well, I agree with the logic you describe in your comment, but I fail to see this in the source code (I might be wrong of course - please elaborate)

 Unless you mean you plan to implement this in the future?

So, recap...

✔️Superuser must be able to send an entirely new email address, mark it a verified and primary at the same time. 
✔️Logic checks should happen on the service. Data integrity/verification checks should happen on the controller. So if a plain user specified an `isVerified` field this should throw an error on the controller. But if a user specifies a non-verified email as primary, then it should throw an error on the service.

Reply:
You are absolutely right. So very sorry this was missing. It is added now. The check apparently was on my mind and never on my code. I also added an extra check on the tests, because i figured out an extra missing case.